### PR TITLE
feat: new requested changes field

### PR DESCRIPTION
**Pull request by emilyjablonski**
# Pull Request Template

## Issue Overview

This PR addresses [#304](https://github.com/metrotranscom/doorway/issues/304)

- [x] This change addresses the issue in full

## Description

Adds new backend fields to support the changes requested flow for listings approval.

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [ ] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: backend/core/src/listings/dto/listing.dto.ts [hunk 1]
```diff
@@ -237,4 +239,10 @@ export class ListingDto extends OmitType(Listing, [
   @Type(() => ListingUtilitiesDto)
   @IsOptional({ groups: [ValidationsGroupsEnum.default] })
   utilities?: ListingUtilitiesDto
+
+  @Expose()
+  @IsOptional({ groups: [ValidationsGroupsEnum.default] })
+  @ValidateNested({ groups: [ValidationsGroupsEnum.default], each: true })
```
Comment (ColinBuyck) [inline backend/core/src/listings/dto/listing.dto.ts:245]: What does the each:true do here?

Comment (emilyjablonski) [reply backend/core/src/listings/dto/listing.dto.ts]: Oop that's for fields that are an array definitely just a copy 🍝  ty!

**Review by ColinBuyck — APPROVED** [review]
LGTM 🐻‍❄️

File: backend/core/src/listings/entities/listing.entity.ts [hunk 2]
```diff
@@ -664,6 +664,27 @@ class Listing extends BaseEntity {
...
+  requestedChanges?: string | null
+
+  @Column({ type: "timestamptz", nullable: true })
+  @Expose()
+  @IsOptional({ groups: [ValidationsGroupsEnum.default] })
+  @IsDate({ groups: [ValidationsGroupsEnum.default] })
+  @Type(() => Date)
+  requestedChangesDate?: Date | null
+
+  @ManyToOne(() => User, { eager: true, cascade: true })
```
Comment (ludtkemorgan) [inline backend/core/src/listings/entities/listing.entity.ts:681]: Do we need the eager and cascade here? 
My understanding is that when "eager" is set to true doing a find on listings will then grab the whole user. We probably won't want the user information being sent unless it has to be.
Also, I don't think we will need to update the user table so cascade is probably not necessary. Unless I'm misunderstanding what those two flags do. @YazeedLoonat might know better

**Pull request merged**